File: backend/app/schemas/ai_assistant.py
from typing import Literal

from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str


class AskRequest(BaseModel):
    question: str = Field(min_length=1, max_length=2000)
    history: list[ChatMessage] = Field(default_factory=list, max_length=20)


class AskResponse(BaseModel):
    answer: str
    tools_used: list[str] = Field(default_factory=list)


class ReindexResponse(BaseModel):
    products_indexed: int

File: frontend/src/api/aiAssistant.ts
import apiClient from "./client";

export interface ChatMessagePayload {
  role: "user" | "assistant";
  content: string;
}

export interface AskResponse {
  answer: string;
  tools_used: string[];
}

export async function askAssistant(
  question: string,
  history: ChatMessagePayload[]
): Promise<AskResponse> {
  const response = await apiClient.post<AskResponse>("/api/v1/ai-assistant/ask", {
    question,
    history,
  });
  return response.data;
}